[Wong Yun Rui Chris] iP

wyrchris opened the pull request:
### DUKE

> _"All quotes are made-up"_ - [Chris Wong](https://github.com/wyrchris) 

Duke is better as it's,

- Text-based
- Fast
- **Fun**😃

How to use it,

1. Download it.
2. Double-click it.

Features:

- [X] Adding/Deleting task
- [X] Marking task as done
- [ ] Searching for task by keyword

Example of the `code`:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

wyrchris committed:
Add level 1

wyrchris committed:
Add level 2

wyrchris committed:
Add level 3

wyrchris committed:
Add level 4

wyrchris committed:
Add Javadocs

wyrchris committed:
Add A-TextUiTesting

wyrchris committed:
Add level 5

wyrchris committed:
Update error handling

wyrchris committed:
Add level 6

wyrchris committed:
Add enums for task

wyrchris committed:
Align with style guide

wyrchris committed:
Add level 7

wyrchris committed:
Add level-8

wyrchris committed:
Merge branch 'branch-Level-7'

wyrchris committed:
Merge branch 'branch-Level-8'

wyrchris committed:
Add  More OOP

wyrchris committed:
Organise into packages

wyrchris committed:
Add JUnit tests

wyrchris committed:
Add JavaDoc comments

wyrchris committed:
Follow the Coding Standard

wyrchris committed:
Add level 9

wyrchris committed:
Merge branch 'branch-A-JavaDoc'

wyrchris committed:
Merge branch 'branch-A-CodingStandard'

wyrchris committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/TaskList.java

wyrchris committed:
Merge remote-tracking branch 'origin/add-gradle-support'

wyrchris committed:
Automate project builds using Gradle

wyrchris committed:
Add a GUI to Duke

wyrchris committed:
Merge branch 'branch-Level-10'

wyrchris committed:
Update Javadocs

Jacobnbh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,63 @@
+            break;
+
+        case "done":
+            // Fallthrough
```
Hello! I like how you made use of `// Fallthrough` here to handle the marking and deleting of tasks. Reduces the number of times the code has to be written (making it succinct) for both types of commands since they require only an index 👍

Jacobnbh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,63 @@
+            break;
+
+        case "todo":
+            // Fallthrough
```
Great use of `// Fallthrough`, same comments as above! 👍

Jacobnbh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,96 @@
+    public void saveData(TaskList list) {
+        try {
+            // get file directory string
+            String[] dir = this.filepath.split("/");
```
I thought `dir` was a little too general for a variable name. Since it's an array, perhaps a plural form could be used? Or naming it differently altogether? 🤔 (e.g. `filePathArray`, even `dirArray` could be better)

Jacobnbh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,96 @@
+ * A class that encapsulates the saving/reading data for Duke.
+ */
+public class Storage {
+    private String filepath;
```
Not sure if "filepath" is one or two words😅 But if it is made up of two words, then as a variable, it should be in CamelCase: `filePath`

Jacobnbh reviewed (commented):
LGTM! Just a few nits to fix. Other than that, code is succinct and easy to understand! 😄

jianoway commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,42 @@
+ *
+ * @author Wong Yun Rui Chris
+ */
+public class Todo extends Task{
```
Possibly redundant suggestion but in accordance with the prof's recommendation for this Task type to be "ToDo", maybe change to the camelcase version?

jianoway commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,59 @@
+     */
+    public Deadline(String description, String by, Boolean isDone) {
+        super(description, isDone);
+        this.by = by;
```
Maybe make this 'by' variable a little more specific? The byLd is the LocalDate version of it so it would be slightly clearer to be more specific with by.

jianoway reviewed (commented):
Overall, one of the cleaner iP repos I've seen. Your implementation of OOP is very good and I'm quite impressed actually!! :D Keep up the good work :)

wyrchris committed:
Refine according to peer reviews

wyrchris committed:
Use CheckStyle

wyrchris committed:
Update code according to error found by checkstyle

wyrchris committed:
Merge branch 'branch-A-CheckStyle'

meerian commented on src/main/java/duke/gui/DialogBox.java:
```diff
@@ -0,0 +1,79 @@
+package duke.gui;
```
Nice use of subpackages!

meerian commented on src/test/java/duke/DukeTest.java:
```diff
@@ -0,0 +1,52 @@
+            System.out.println(e.getMessage());
+        }
+    }
+}
```
I like how rigorous your JUnit testings are!

meerian reviewed (commented):
Good job! Keep up the good work!

wyrchris committed:
Use Assertions

Need to ensure that certains assumption during the execution of the code is held  to be true.
The TaskList assumes that the input passed to it is not blank and this must be held true to
ensure that the correct method is called with the correct input.

Let's use assert statement in the TaskList methods that is called by the Parser handleInput
method to check for the correctness of the input that is passed to it.

It is best for the individual method to check for the correct input as the input requirement is
not the same between them.

wyrchris committed:
Improve code quality

Numerous portion of the code is of different level of abstraction and to streamline the various
methods to use a single level of abstraction as much as possible.

Let's extract out various method from code which are too long.

This is to simplify the varioous code and to be more friendly to read.

wyrchris committed:
Merge pull request #1 from wyrchris/branch-A-CodeQuality

Improve code quality

wyrchris committed:
Merge branch 'master' of https://github.com/wyrchris/ip

wyrchris committed:
Merge branch 'master' into branch-A-Assertions

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java

wyrchris committed:
Merge pull request #2 from wyrchris/branch-A-Assertions

Use Assertions

wyrchris committed:
Merge branch 'master' of https://github.com/wyrchris/ip

wyrchris committed:
Provide a way to undo and redo a command

Better greater experience for the user to be able undo a mistake.

Let's implement undo and redo by keeping track of previous states that we can revert
the TaskList back to.

By keeping track of all the previous state, this will allow the user to undo/redo multiple
commands back or forth.

wyrchris committed:
Merge branch 'branch-C-Undo'

wyrchris committed:
Add FXML support for JavaFX code

wyrchris committed:
Improve the GUI

wyrchris committed:
Add a User Guide
